docs/development/pigs/pig-011.rst

```diff
@@ -0,0 +1,274 @@
+absence, of a ``RegionGeom`` we could simply expect a reco energy and true energy
+``MapAxis``.
+
+We do not detail possible approaches in the current PIG. These should be re-evaluated
+in the more general context of data reduction. Some examples, using the current
+data reduction approach will be exposed to users in a dedicated notebook.
+
+Both approaches will result in a list of ``Datasets`` consisting of ``MapDataset``
+or ``SpectrumDatasetOnOff`` with identical geometries for each time bin.
+
+In order to properly assign times to any ``Dataset``, the latter must therefore carry
+the time information. Minimally, this can be done with a meta information such as
+``time_start`` and ``time_stop``. This does not give a full idea of the coverage
+of time bin though. Ideally, the ``Dataset`` should track the ``GTI`` table of
+the filtered ``Eventlist`` that were used for its production. It can be stored on the
+``Dataset`` itself and be serialized independently or be added as an extension to
+the serialized ``count`` data container (a ``CountsSpectrum`` or ``Map``), as is
+done for OGIP spectra. In order to stack several ``Dataset`` objects it is necessary
+to be able to combine ``GTI`` tables. While a simple stacking of tables is enough
+in many cases, the situation of overlapping time intervals should be considered and
+a proper ``GTI.union()`` should be introduced (a functionality recurrent in HEA
+software see e.g. ftools `mgtime)``
+
+
+Data Fitting
+------------
+
+The data fitting is the step were the ``Datasets`` are converted into a lightcurve
+based on a model of the emission. The amplitude of the source model is fitted while
+other parameters are usually, but not necessarily, fixed.
+
+The fitting is very similar to the extraction of ``FluxPoints`` in the spectral analysis
+and does not strongly depend on the type of ``Datasets`` considered. The new
+:class:``LightCurveEstimator`` class should therefore be able to take as input both
+``SpectrumDatasetOnOff`` and ``MapDataset``, in a similar fashion as the current
+``FluxPointEstimator``.
+
+After creating the ``Datasets``, the user would first define the model to be used
+and freeze its parameters. Then we would apply it to the various ``Dataset`` objects
+before calling the ``LightCurveEstimator``:
+
+.. code ::
+
+    sky_model = SkyModel(spatial_model=spatial_model, spectral_model=spectral_model)
+    sky_model.parameters["index"].frozen = True
+    sky_model.parameters["lon_0"].frozen = True
+    sky_model.parameters["lat_0"].frozen = True
+
+    for dataset in datasets:
+        dataset.model = sky_model
+
+    lc_estimator = LightCurveEstimator(datasets)
+    lightcurve = lc.run()
+
+To help, the user deal with model parameters some helper function could be introduced
+to freeze all parameters of a model. Another possibility is to assume that all parameters
+are fixed except the `amplitude` in the ``LightCurveEstimator.run()`` method, or pass as
```

Comment:
For the `FluxPointEstimator` I have implemented a `reoptimize` argument. The basic logic is as following:
- With `reoptimize=False` only the amplitude of the spectral model is fitted.
- With  `reoptimize=True` all the free parameters of the dataset on init are fitted. This means the user specifies which parameters are frozen or not directly on the dataset objects (as for a "normal" likelihood fit).
I think this probably the simplest solution for now.